File: src/main/clojure/co/gywb/demo/web_front/clojure/hello.clj
(ns co.gywb.demo.web-front.clojure.hello
  (:import
    [javax.ws.rs GET Path PathParam Produces]))

(defn greet-impl [name]
  (str "Hello, " name ", from Clojure!"))

(gen-class
  :name
  ^{Path "/hello/clojure"} co.gywb.demo.web_front.clojure.Hello
  :state state
  :prefix "-"
  :methods [[^{GET true
               Path "/{name}"
               Produces ["text/plain"]}
             greet [^{PathParam "name"} String] String]])

(defn -greet [_ ^String name]
  (greet-impl name))

File: src/main/java/co/gywb/demo/web_front/java/Hello.java
package co.gywb.demo.web_front.java;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/hello/java")
public class Hello {

    @GET
    @Path("/{name}")
    @Produces(MediaType.TEXT_PLAIN)
    public String greet(@PathParam("name") String name) {
        return "Hello, " + name + ", from Java!";
    }
}
